File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode start = new ListNode();
        start.next = head;
        ListNode s = start;
        ListNode f = start;
        int k = 0;
        while(k < n){
            s = s.next;
            k++;
        }
        while(s.next != null){
            s = s.next;
            f = f.next;
        }
        f.next = f.next.next;
        return start.next;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* start = new ListNode();
        start->next = head;
        ListNode* p = start;
        ListNode* q = start;
        int k = 0;
        while(k < n){
            p = p->next;
            k++;
        }
        while(p->next != NULL){
            p = p->next;
            q = q->next;
        }
        q->next = q->next->next;
        return start->next;
    }
};